apps/omg_watcher_info/lib/omg_watcher_info/api/transaction.ex

```diff
@@ -107,28 +108,100 @@ defmodule OMG.WatcherInfo.API.Transaction do
     end
   end
 
+  @doc """
+  Converts parameter keyword list to a map before passing it to multi-clause "handle_merge/`1"
+  """
+  @spec merge(Keyword.t()) :: create_t()
+  def merge(parameters) do
+    parameters |> Map.new() |> handle_merge()
+  end
+
+  @spec handle_merge(map()) :: create_t()
+  defp handle_merge(%{address: address, currency: currency}) do
+    merge_inputs =
+      address
+      |> DB.TxOutput.get_sorted_grouped_utxos(:asc)
+      |> Map.get(currency, [])
+
+    case merge_inputs do
+      [] ->
+        {:error, :no_inputs_found}
+
+      [_single_input] ->
+        {:error, :single_input}
+
+      inputs ->
+        {:ok, TransactionCreator.generate_merge_transactions(inputs)}
+    end
+  end
+
+  defp handle_merge(%{utxo_positions: utxo_positions}) do
+    with {:ok, inputs} <- get_merge_inputs(utxo_positions),
+         :ok <- no_duplicates(inputs),
+         :ok <- single_owner(inputs),
+         :ok <- single_currency(inputs) do
+      {:ok, TransactionCreator.generate_merge_transactions(inputs)}
+    end
+  end
+
   defp get_utxos_count(inputs) do
     inputs
-      |> Enum.map(fn {_, utxos} -> utxos end)
-      |> List.flatten()
-      |> Enum.count
+    |> Enum.map(fn {_, utxos} -> utxos end)
+    |> List.flatten()
+    |> Enum.count()
   end
 
   defp create_transaction(utxos_count, inputs, _order) when utxos_count > Transaction.Payment.max_inputs() do
     inputs
-      |> TransactionCreator.generate_merge_transactions()
-      |> respond(:intermediate)
+    |> TransactionCreator.generate_merge_transactions()
+    |> respond(:intermediate)
   end
 
   defp create_transaction(_utxos_count, inputs, order) do
     inputs
-      |> TransactionCreator.create(order)
-      |> respond(:complete)
+    |> TransactionCreator.create(order)
+    |> respond(:complete)
+  end
+
+  @spec get_merge_inputs(list()) :: {:ok, list()} | {:error, atom()}
+  defp get_merge_inputs(utxo_positions) do
+    Enum.reduce_while(utxo_positions, {:ok, []}, fn encoded_position, {:ok, acc} ->
+      case encoded_position |> Utxo.Position.decode!() |> DB.TxOutput.get_by_position() do
```

Comment:
This could be optimised to query DB only once and return all inputs needed, but it's not a big deal as we're limiting the amount of `utxo_positions` to 4.